### PR TITLE
Support account alias in rest API

### LINKED ISSUE #2964: Expose alias and add alias filter to /accounts
### Problem

[HIP-32](https://hips.hedera.com/hip/hip-32) adds aliases to accounts.  Users will need a way to find their actual account id from a given alias.

### Solution

- Expose `alias` on `/api/v1/accounts`
- Add an `alias` filter to `/api/v1/accounts` to search for accounts by a given alias.  The filter should support three formats, `0.0.x`, `0.x`, and `x`, where x is a base32 encoded key.  Input will need to be converted to hex or raw bytes in order to search the database.

### Alternatives

_No response_

### PR DESCRIPTION
**Pull request by xin-hedera**
**Description**:

This PR adds account alias support in rest API:

- Add RFC4648 no padding base32 codec
- Add account alias parser
- Add accountAlias path param to `/v1/api/accounts/:accountAliasOrAccountId`
- Add `alias` to account object in json response
- Add new response code `UNEXPECTED_TOKEN_DECIMALS`

**Related issue(s)**:

Fixes #2964

**Notes for reviewer**:

Link to the previous PR: #3087

**Checklist**

- [x] Documented (Code comments, README, etc.)
- [x] Tested (unit, integration, etc.)

### CODE REVIEW

File: hedera-mirror-rest/__tests__/config.test.js [hunk 1]
```diff
@@ -52,7 +52,7 @@ beforeEach(() => {
 });
 
 afterEach(() => {
-  fs.rmdirSync(tempDir, {recursive: true});
+  fs.rmSync(tempDir, {recursive: true});
```
Comment (xin-hedera) [inline hedera-mirror-rest/__tests__/config.test.js:55]: fs.rmdirSync is marked as deprecated.

File: hedera-mirror-rest/__tests__/specs/accounts-07-not-found.spec.json [hunk 2]
```diff
@@ -53,7 +57,15 @@
     "transactions": [],
     "cryptotransfers": []
   },
-  "url": "/api/v1/accounts/0.0.7",
+  "urls": [
+    "/api/v1/accounts/7",
+    "/api/v1/accounts/0.7",
+    "/api/v1/accounts/0.0.7",
+    "/api/v1/accounts/AAAAAA5L3ZGSRVUCSPDR7TONZSRY3D5OMEBKQMVTD2AC6JL72HMQ",
```
Comment (xin-hedera) [inline hedera-mirror-rest/__tests__/specs/accounts-07-not-found.spec.json:64]: non-existing alias, expect 404

File: hedera-mirror-rest/__tests__/specs/accounts-07-not-found.spec.json [hunk 3]
```diff
@@ -53,7 +57,15 @@
...
+    "/api/v1/accounts/KGNABD5L3ZGSRVUCSPDR7TONZSRY3D5OMEBKQMVTD2AC6JL72HMQ",
+    "/api/v1/accounts/0.KGNABD5L3ZGSRVUCSPDR7TONZSRY3D5OMEBKQMVTD2AC6JL72HMQ",
+    "/api/v1/accounts/0.0.KGNABD5L3ZGSRVUCSPDR7TONZSRY3D5OMEBKQMVTD2AC6JL72HMQ"
```
Comment (xin-hedera) [inline hedera-mirror-rest/__tests__/specs/accounts-07-not-found.spec.json:67]: alias maps to a deleted account, also expect 404

File: hedera-mirror-rest/__tests__/specs/accounts-23-alias-multiple-match.spec.json [hunk 4]
```diff
@@ -0,0 +1,41 @@
+{
+  "description": "Account api calls for specific alias matching multiple alive accounts",
```
Comment (steven-sheehy) [inline hedera-mirror-rest/__tests__/specs/accounts-23-alias-multiple-match.spec.json:2]: Should probably have a spec where multiple match but only one is not deleted

File: hedera-mirror-rest/base32.js [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * ‍
+ */
+
+'use strict';
+
+const {base32} = require('rfc4648');
+
+const decodeOpts = {loose: true};
```
Comment (steven-sheehy) [inline hedera-mirror-rest/base32.js:25]: Why loose?

Comment (xin-hedera) [reply hedera-mirror-rest/base32.js:25]: padding. loose accepts non-padding base32 strings, but it does more:

```
      opts.loose
        ? string
            .toUpperCase()
            .replace(/0/g, 'O')
            .replace(/1/g, 'L')
            .replace(/8/g, 'B')
```

File: hedera-mirror-rest/__tests__/specs/accounts-23-alias-multiple-match.spec.json (hunk 4 above)
Comment (xin-hedera) [reply hedera-mirror-rest/__tests__/specs/accounts-23-alias-multiple-match.spec.json:2]: accounts-10-account-transactions.spec.json covers it:

```json
  "extendedDescription": [
    "account 8 and account 9 have the same alias, however, account 9 is deleted, so when querying by the alias, ",
    "the rest API should return the information for account 8"
  ],
```

**Review by steven-sheehy — DISMISSED** [review]
LGTM

**Review by steven-sheehy — APPROVED** [review]

**Review by ijungmann — APPROVED** [review]

**Pull request merged**